server/src/OutputParser.ts

```diff
@@ -14,13 +14,17 @@ import {
   ProblemsContainer
 } from './ProblemsContainer'
 
+let _connection: Connection | null = null
```

Comment:
this makes it impossible for the server to handle multiple clients at once?
If we want to go that fat, this should be an array stored in the server and passed to this class.

Reply:
[Here](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#languageServerProtocol)'s what the current specification seems to says about having multiple clients at one:
> The protocol currently assumes that one server serves one tool. There is currently no support in the protocol to share one server between different tools. Such a sharing would require additional protocol e.g. to lock a document to support concurrent editing.

Actually, I noticed yesterday the method [onWillSaveWaitUntil](https://github.com/microsoft/vscode-languageserver-node/blob/84285312d8d9f22ee0042e709db114e5415dbdde/server/src/common/textDocuments.ts#L123) of TextDocuments that seems to be intended for the case of concurrent editings: 
> Sets a handler that will be called if a participant wants to provide edits during a text document save.

Reply:
As discussed, let's keep what it is for now.